castle/cms/browser/content/__init__.py

```diff
@@ -186,12 +186,19 @@ def chunk_upload(self):
             if not os.path.exists(info['tmp_file']):
                 raise Exception('No tmp upload file found')
         fi = open(info['tmp_file'], mode)
-
-        while True:
-            data = self.request.form['file'].read(2 << 16)
-            if not data:
-                break
-            fi.write(data)
+        type_, location = self.get_type_and_location(self.request.form)
+        if type_ == 'Image':
```

Comment:
maybe instead of duplicating the code between if/else blocks, this could be condensed down to a single block like the removed code, except with the form[] key value being dynamic

eg

```
formtype = 'file'
if type_ == 'Image':
    formtype = 'image'

# ...
data = self.request.form[formtype].read(2 << 16)
# ..
```